aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -88,12 +86,10 @@ public OutputFile newOutputFile(String path) {
   @Override
   public void deleteFile(String path) {
```

Comment:
Is there a guarantee that the path isn't a "directory"?

Reply:
There's really no concept of a directory (in the sense that you can recursively delete a bunch of data) in S3.  Directories are just a byproduct of whatever you use as a delimiter in listing (most typically `/` to make it look like a filesystem structure).  Some FileSystem implementation use a marker for empty directories (e.g. `<path>_$folder$`), but that's just to workaround the fact that there isn't such a thing.

For Iceberg, this should be safe as listing isn't used and files are all directly referenced.